armada_jupyter/armada.py

```diff
@@ -0,0 +1,69 @@
+"""
+Example of getting jupyter notebook running on a k8s cluster.
+"""
+
+
+import grpc
+from armada_client.client import ArmadaClient
+from armada_client.armada import submit_pb2
+
+from armada_jupyter.constants import HOST, PORT, DISABLE_SSL
+
+
+def create_jupyter_service():
+    """
+    Create a jupyter service.
+    """
+
+    return submit_pb2.ServiceConfig(
+        ports=[8888],
+        type=submit_pb2.ServiceType.NodePort,
+    )
+
+
+def create_jupyter_ingress():
+    """
+    Create a jupyter ingress.
+    """
+
+    return submit_pb2.IngressConfig(ports=[8888], tls_enabled=False, use_clusterIP=True)
+
+
+def create_armada_request(submission, client):
+    priority = submission.armada_priority
+
+    # Create the PodSpec for the job
+    return client.create_job_request_item(
+        priority=priority,
+        pod_spec=submission.to_podspec(),
+        ingress=[create_jupyter_ingress()],
+        services=[create_jupyter_service()],
+        labels={"app": "jupyter"},
+    )
+
+
+def submit(submission):
+    """
+    Starts a workflow for jupyter notebook.
+    """
+
+    # Ensures that the correct channel type is generated
+    if DISABLE_SSL:
+        channel = grpc.insecure_channel(f"{HOST}:{PORT}")
+    else:
+        channel_credentials = grpc.ssl_channel_credentials()
+        channel = grpc.secure_channel(
+            f"{HOST}:{PORT}",
+            channel_credentials,
+        )
+
+    client = ArmadaClient(channel)
+
+    queue = submission.armada_queue
+    job_set_id = "armada-jupyter-pods"
+
+    # Create the PodSpec for the job
+    job_request_item = create_armada_request(submission, client)
+    client.submit_jobs(
```

Comment:
When it comes to unit testing things like this, I like to have this logic refacted into a function.  

For your unit test, you could have client.submit_jobs be a dummy function so you could test this entire function minus the submit_jobs call.  

I'm okay with you holding off on that now but just a suggestion when you want to write more tests.

Reply:
Just for speed, maybe creating an issue for this and track it that way for now is okay?